### DISCUSSION TITLE
Question about a syntax in THM

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#22865] [Q&A General] [answered]
I'm learning the [HeatTransferFromHeatStructure1Phase.C](https://github.com/idaholab/moose/blob/d25aa445f6a00c23e22d343095e8e7999ba46714/modules/thermal_hydraulics/src/components/HeatTransferFromHeatStructure1Phase.C) and there's a syntax I don't know the meaning:
`params.set<BoundaryName>("secondary_boundary") = {getSlaveSideName()};`
Why there is a {}?
There are some codes like:
`params.set<std::vector<BoundaryName>>("boundary") = {getSlaveSideName()};`
`params.set<std::vector<BoundaryName>>("boundary") = {getMasterSideName()};`
Can anybody tell me the effect of the {}?

**Comment by GiudGiud** [comment, marked as answer]
Hello

{} creates a C++ standard vector with the items inside the bracket contained in the vector. The parameter has not been added as a single instance, it needs to be a vector 

Guillaume

**Reply by Ethan-xj** [reply]
Oh I got it. Thanks.
I actually noticed that the parameter type before every {xxx} is vector. And there is another point that confused me:
What is the type `BoundaryName`? Can a vector be BoundaryName?
The 185 line shows:
`params.set<BoundaryName>("secondary_boundary") = {getSlaveSideName()};`

**Reply by GiudGiud** [reply]
BoundaryName is an alias for a string
You can have a vector of boundary names
{"boundary_1", "boundary_2"} for example

**Reply by Ethan-xj** [reply]
I understand. Thank you :D
